axelor-stock/src/main/java/com/axelor/apps/stock/web/PickedProductsController.java

```diff
@@ -0,0 +1,33 @@
+package com.axelor.apps.stock.web;
+
+import com.axelor.apps.base.AxelorException;
+import com.axelor.apps.stock.db.PickedProducts;
+import com.axelor.apps.stock.db.repo.PickedProductsRepository;
+import com.axelor.apps.stock.service.PickedProductsService;
+import com.axelor.inject.Beans;
+import com.axelor.rpc.ActionRequest;
+import com.axelor.rpc.ActionResponse;
+
+public class PickedProductsController {
+
+  public void realizePicking(ActionRequest request, ActionResponse response)
+      throws AxelorException {
+    PickedProducts pickedProducts = request.getContext().asType(PickedProducts.class);
+    pickedProducts = Beans.get(PickedProductsRepository.class).find(pickedProducts.getId());
+    try {
+      Beans.get(PickedProductsService.class)
+          .createStockMoveAndStockMoveLine(pickedProducts.getMassStockMove(), pickedProducts);
+    } catch (IllegalStateException e) {
+      response.setAlert(e.getMessage());
```

Comment:
Any exception at this point should be traced.
TraceBackService.trace(response, e)